### PR TITLE
Use resolve for OG metadata

### LINKED ISSUE #6464: Use resolve for OG metadata instead of chainquery
The chainquery code is flaky and CQ data may be wrong at time + gets hung up. We should just use resolve for this.

### LINKED ISSUE #6780: GoogleVideo Metadata: "uploadDate" tag incorrect
We are currently setting the wrong release date for some (maybe most) of the videos, causing them to appear as "uploaded on 1970".  Full thread [here](https://lbryians.slack.com/archives/C81FGKR51/p1628040044020000).

CQ currently doesn't return `creation_timestamp`.  The suggested use of "transaction time" is not reliable either, as it represents the latest (for edits?) transaction time, so it is neither matches the "creation" nor "release" time in the GUI.

Wait for:   [Use resolve for OG metadata instead of chainquery #6464](https://github.com/lbryio/lbry-desktop/issues/6464)

### LINKED ISSUE #6812: Inconsistent url redirection on development mode
**Bug**

**To Reproduce**
1. Go to https://odysee.com/cat/3b89edbe2aac27ae9cb8bb09b4d837882e827cf4
2. URL is redirected to https://odysee.com/@CatsAreTheBest:0/cat:7?

It should redirect to https://odysee.com/@tobban:e/cat:3

**Expected behavior**
Claim id is ignored if its added on urls after a `/` instead of `#` or `:`.
This affects the embed api  or other similar urls.

### PR DESCRIPTION
**Pull request by btzr-io**
## PR Checklist

Please check all that apply to this PR using "x":

- [x] I have checked that this PR is not a duplicate of an existing PR (open, closed or merged)
- [x] I added a line describing my change to CHANGELOG.md
- [ ] I have checked that this PR does not introduce a breaking change
- [ ] This PR introduces breaking changes and I have provided a detailed explanation below

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting)
- [x] Refactoring (no functional changes)
- [ ] Documentation changes
- [ ] Other - Please describe:

## Fixes

Closes #6780
Closes #6464
Closes #6812

### CODE REVIEW

**Comment by btzr-io** [comment]
I'm still not sure how to test this...

**Comment by infinite-persistence** [comment]
- `yarn && yarn dev:web-server`
- `localhost:1337` in the browser after build is done.
- Open a content page, then F5 to reload (or enter URL directly).  The OG is not populated on link navigation.
- Verify the `<head>` for the OG and GoogleVideo info 
  -

**Comment by btzr-io** [comment]
> 
> 
>     * `yarn && yarn dev:web-server`
> 
>     * `localhost:1337` in the browser after build is done.
> 
>     * Open a content page, then F5 to reload (or enter URL directly).  The OG is not populated on link navigation.
> 
>     * Verify the `<head>` for the OG and GoogleVideo info
>       
>       * 

Nice thanks ✌️

File: web/src/html.js [hunk 1]
```diff
@@ -242,24 +264,25 @@ function buildGoogleVideoMetadata(claimUri, claim) {
...
-
-  const rows = await getClaim(claimName, claimId, channelName, channelClaimId);
-  if (rows && rows.length) {
-    const claim = rows[0];
-
-    if (claim.reposted_name && claim.reposted_claim_id && !ignoreRedirect) {
-      ctx.redirect(`/${claim.reposted_name}:${claim.reposted_claim_id}`);
-      return;
+  try {
+    const url = `lbry://${claimName}:${claimId}`;
```
Comment (btzr-io) [inline web/src/html.js:273]: `:` or `#` ?

Comment (btzr-io) [reply web/src/html.js:273]: Looks like ":" also works now.

File: web/src/html.js [hunk 2]
```diff
@@ -122,20 +127,33 @@ function buildBasicOgMetadata() {
...
+  const { meta, value, signing_channel } = claim;
+  const fee = value && value.fee && (Number(value.fee.amount) || 0);
+  const tags = value && value.tags;
+  const media = value && (value.video || value.audio || value.image);
+  const source = value && value.source;
+  const channel = signing_channel && signing_channel.name;
+  const thumbnail = value && value.thumbnail && value.thumbnail.url;
+  const mediaType = source && source.media_type;
+  const mediaDuration = media && media.duration;
+  const creationTimestamp = meta && (meta.creation_timestamp || 0);
```
Comment (infinite-persistence) [inline web/src/html.js:139]: Should do `value.release_time --> creation time --> 0`. We've been using release time, which works for newer claims.

**Comment by infinite-persistence** [comment]
Nice, seems like "keyword"/tags now work too.  I'm guessing CQ wasn't returning tags all this while.
image

File: CHANGELOG.md [hunk 3]
```diff
@@ -18,6 +18,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - Improve twitter share _community pr!_ ([#6690](https://github.com/lbryio/lbry-desktop/pull/6690))
 - Update lighthouse search api _community pr!_ ([#6731](https://github.com/lbryio/lbry-desktop/pull/6731))
 - Update sockety api _community pr!_ ([#6747](https://github.com/lbryio/lbry-desktop/pull/6747))
+- Use resolve for OG metadata instead of chainquery _community pr!_ ([#6787](https://github.com/lbryio/lbry-desktop/pull/6787))
```
Comment (infinite-persistence) [inline CHANGELOG.md:21]: Not sure where the error is, but Embed OG's aren't showing up.
Example embed: https://odysee.com/$/embed/eevblog-1412-argon-ion-laser-teardown!/1fe325659bf8d791f72e02bb614a942fa4b571bf?r=5yqd2A3cQ59DhkNRYTJq3BApzdcr7KSn

Comment (btzr-io) [reply CHANGELOG.md:21]: The embed url doesn't use `:` or `#` to separate the id.

Comment (btzr-io) [reply CHANGELOG.md:21]: Adding this `.replace('\/', '#');` fixes the issue. Not sure if this will work on all cases.

Comment (btzr-io) [reply CHANGELOG.md:21]: As long as embed url is always `$/embed/name/id`  ^^

**Comment by btzr-io** [comment]
Is there any other urls that uses `/` instead of `:` or `#` before the claim id ?

**Comment by infinite-persistence** [comment]
> _Is there any other urls that uses / instead of : or # before the claim id ?_

Tom mentioned that anywhere we use `generateStreamURL`, `generateDownloadUrl` and `generateDirectUrl`.

I noticed that `http://odysee.com/eevblog-1412-argon-ion-laser-teardown!/1fe325659bf8d791f72e02bb614a942fa4b571bf` redirects and shows the right header, but the PR version does not.

**Comment by btzr-io** [comment]
> However, I noticed that `http://odysee.com/eevblog-1412-argon-ion-laser-teardown!/1fe325659bf8d791f72e02bb614a942fa4b571bf` redirects and shows the right header, but the PR version does not.

Same issue with embed urls. parseURI fails to detect the claim id.

**Comment by btzr-io** [comment]
> http://odysee.com/eevblog-1412-argon-ion-laser-teardown!/1fe325659bf8d791f72e02bb614a942fa4b571bf 

This is actually the equivalent of:
> lbry://eevblog-1412-argon-ion-laser-teardown!

Instead of:
> lbry://eevblog-1412-argon-ion-laser-teardown!#1fe325659bf8d791f72e02bb614a942fa4b571bf

**Comment by btzr-io** [comment]
> However, I noticed that http://odysee.com/eevblog-1412-argon-ion-laser-teardown!/1fe325659bf8d791f72e02bb614a942fa4b571bf redirects and shows the right header, but the PR version does not.

I don't think the redirection issue has nothing todo with the changes on this pull request.

**Pull request merged**